direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -790,6 +819,6 @@ private WatermarkEstimator createWatermarkEstimator() {
   }
 
   private static PartitionedWatermarkEstimator createWatermarkEstimatorForEmptyParts() {
-    return (PartitionedWatermarkEstimator) System::currentTimeMillis;
+    return () -> Watermarks.MAX_WATERMARK;
```

Comment:
MIN_WATERMARK should be safer

Reply:
This is not about 'safer' but 'correct'. When there are no partitions, there is no data and therefore it is end-of-input. It is needed for the overall functionality.